crates/astria-sequencer/src/mempool/mod.rs

```diff
@@ -184,11 +180,18 @@ impl Mempool {
                 // Release the lock asap.
                 drop(pending);
                 // try to add to parked queue
-                parked.add(
-                    timemarked_tx,
+                match parked.add(
+                    timemarked_tx.clone(),
```

Comment:
i think you can do `let id = timemarked_tx.id()` before this, then use that later on in the function and remove the `clone()`